src/witan/workspace_api.clj

```diff
@@ -48,30 +49,20 @@
 (defmacro merge->
   "Macro sending x to multiple forms and then merging the results
   TODO: Make this way more resiliant to inline functions and other macros (such as threading)"
-  [x & forms]
-  (let [split (map (fn [f]
-                     (if (seq? f)
-                       (vector (first f) (-> f rest vec))
-                       (vector f []))) forms)]
-    `(apply merge (map (fn [[f# args#]] (apply f# ~x args#)) (list ~@split)))))
+  [data & forms]
+  (loop [forms forms, result `(apply merge)]
+    (if forms
+      (let [form (first forms)
+            result (concat result (list `(-> ~data ~form)))]
+        (recur (next forms) result))
+      result)))
```

Comment:
You might be able to do this a bit more like:

`(apply merge ~@(map list forms (constantly data)))